internal/callbacker/sender_test.go

```diff
@@ -0,0 +1,161 @@
+package callbacker_test
+
+import (
+	"bytes"
+	"encoding/json"
+	"log/slog"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/bitcoin-sv/arc/internal/callbacker"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestCallbackSender_Send(t *testing.T) {
+	tests := []struct {
+		name            string
+		responseStatus  int
+		expectedSuccess bool
+		expectedRetries int
+	}{
+		{
+			name:            "success - callback sent",
+			responseStatus:  http.StatusOK,
+			expectedSuccess: true,
+			expectedRetries: 1,
+		},
+	}
+
+	for _, tc := range tests {
+		t.Run(tc.name, func(t *testing.T) {
+			// Given
+			retries := 0
+			server := httptest.NewServer(http.HandlerFunc(func(w http.ResponseWriter, _ *http.Request) {
+				w.WriteHeader(tc.responseStatus)
+				retries++
+			}))
+			defer server.Close()
+			logger := slog.New(slog.NewTextHandler(new(bytes.Buffer), nil))
+			sender, err := callbacker.NewSender(server.Client(), logger)
+			require.NoError(t, err)
+			defer sender.GracefulStop()
+
+			// When
+			success := sender.Send(server.URL, "test-token", &callbacker.Callback{TxID: "1234", TxStatus: "SEEN_ON_NETWORK"})
+
+			// Then
+			require.Equal(t, tc.expectedSuccess, success)
+			require.Equal(t, tc.expectedRetries, retries)
+		})
+	}
+}
+
+func TestCallbackSender_GracefulStop(t *testing.T) {
+	// Given
+	logger := slog.New(slog.NewTextHandler(new(bytes.Buffer), nil))
+	sender, err := callbacker.NewSender(http.DefaultClient, logger)
+	require.NoError(t, err)
+
+	// When: Call GracefulStop twice to ensure it handles double stop gracefully
+	sender.GracefulStop()
+	sender.GracefulStop()
+
+	// Then: Assert that sender is disposed
+	err = sender.Health()
+	require.EqualError(t, err, "callbacker is disposed already")
+}
+
+func TestCallbackSender_Health(t *testing.T) {
+	// Given
+	logger := slog.New(slog.NewTextHandler(new(bytes.Buffer), nil))
+	sender, err := callbacker.NewSender(http.DefaultClient, logger)
+	require.NoError(t, err)
+
+	// When: Test Health on active sender
+	err = sender.Health()
+
+	// Then: Expect no error when sender is active
+	require.NoError(t, err)
+
+	// When: Stop sender and check Health error
+	sender.GracefulStop()
+	err = sender.Health()
+
+	// Then: Expect error indicating the sender is disposed
+	require.EqualError(t, err, "callbacker is disposed already")
+}
+
+func TestCallbackSender_SendBatch(t *testing.T) {
+	tests := []struct {
+		name            string
+		url             string
+		token           string
+		dtos            []*callbacker.Callback
+		responseStatus  int
+		expectedSuccess bool
+	}{
+		{
+			name:            "success - batch sent",
+			url:             "/batch",
+			token:           "test-token",
+			dtos:            []*callbacker.Callback{{TxID: "1234", TxStatus: "SEEN_ON_NETWORK"}, {TxID: "5678", TxStatus: "MINED"}},
+			responseStatus:  http.StatusOK,
+			expectedSuccess: true,
+		},
```

Comment:
Add another case where `responseStatus` is an error e.g. `http.StatusInternalServerError`